modulus/utils/graphcast/graph.py

```diff
@@ -40,47 +44,33 @@ class Graph:
 
     Parameters
     ----------
-    icospheres_path : str
-        Path to the icospheres json file.
-        If the file does not exist, it will try to generate it using PyMesh.
     lat_lon_grid : Tensor
         Tensor with shape (lat, lon, 2) that includes the latitudes and longitudes
         meshgrid.
+    multimesh_level: int, optional
+        Level of the multi-mesh, by default 6
     dtype : torch.dtype, optional
         Data type of the graph, by default torch.float
     """
 
     def __init__(
-        self, icospheres_path: str, lat_lon_grid: Tensor, dtype=torch.float
+        self, lat_lon_grid: Tensor, multimesh_level=6, dtype=torch.float
     ) -> None:
         self.dtype = dtype
-        # Get or generate the icospheres
-        try:
-            with open(icospheres_path, "r") as f:
-                loaded_dict = json.load(f)
-                icospheres = {
-                    key: (np.array(value) if isinstance(value, list) else value)
-                    for key, value in loaded_dict.items()
-                }
-                logger.info(f"Opened pre-computed graph at {icospheres_path}.")
-        except FileNotFoundError:
-            from modulus.utils.graphcast.icospheres import (
-                generate_and_save_icospheres,
-            )
-
-            logger.info(
-                f"Could not open {icospheres_path}...generating mesh from scratch."
-            )
-            generate_and_save_icospheres()
-
-        self.icospheres = icospheres
-        self.max_order = (
-            len([key for key in self.icospheres.keys() if "faces" in key]) - 2
-        )
 
         # flatten lat/lon gird
         self.lat_lon_grid_flat = lat_lon_grid.permute(2, 0, 1).view(2, -1).permute(1, 0)
 
+        # create the multi-mesh
+        _meshes = get_hierarchy_of_triangular_meshes_for_sphere(splits=multimesh_level)
+        merged_mesh = merge_meshes(_meshes)
```

Comment:
should be add a config-option specifying whether we want to merge meshes?
e.g. GenCast has
> Unlike the multimesh used in GraphCast, the mesh in GenCast is a 6-times refined icosahedral mesh as
defined in Lam et al. (2023), with 41,162 nodes and 246,960 edge

which on the other hand only works due to the use of self-attention on a 32-hop neighborhood
> The 4-head self-attention mechanism in each block is such that each
node in the mesh attends to itself and to all other nodes which are within its 32-hop neighbourhood
on the mesh

on a second thought, the latter would require no "merging" step but adding these k-hops neighborhoods explicitly anyways, so probably a bit more complicated anyways

feel free to ignore this comment or add further comments :D

Reply:
How about we add this comment to our [epic](https://github.com/NVIDIA/modulus/issues/499) and revisit it if/when we decided to extend to GenCast?

Reply:
makes perfect sense, yes